### ISSUE TITLE
Support formatting ASP.NET vNext projects

### ISSUE THREAD

**Issue by conniey** [dotnet/codeformatter#87]
I have an library that is written in C#, but I converted it to a .kproj (ASP.NET vNext library) so that my ASP.NET vNext website could also use its logic. 

CodeFormatter skips over this project and doesn't format it. so I have to manually create a .csproj, add the .cs files, and run the formatter tool again.

(.kroj Samples)[https://github.com/aspnet/XRE/tree/dev/samples].

**Comment by jaredpar** [comment]
@jasonmalinowski is there any way to load up ASP.Net vNext projects in Roslyn today?  Right now the CodeFormatter just uses a MSBuildWorkspace.  If there is an easy way to load this project type I'm sure it wouldn't take much work to get the code formatter to support it.

**Comment by davidfowl** [comment]
It's possible but not very nice right now. We're working on an API that will make this less sucky.

**Comment by jaredpar** [comment]
@davidfowl is there a GitHub issue I can track for this API?  Once it's available I'd like to try it out here.

**Comment by davidfowl** [comment]
Not really, we're mostly layering things differently (some in NuGet, some in XRE) to make this possible. I'll update this thread when we get closer to having something useful.

**Comment by danroth27** [comment]
@conniey Please also note that you don't have to convert your existing library project to a kproj to be able to reference it from your ASP.NET 5 app as long as you are ok with the app running on the full .NET Framework. You can reference csproj based projects from an ASP.NET 5 app. You would need to port your existing project only if you wanted to be able to run on .NET Core.

**Comment by Petermarcu** [comment]
@conniey @danroth27 We also plan on adding ASP.NET 5 as a target from a Portable Library so you wont need to have a seperate project file and multiple builds if you want to have you're shared logic applicable to multiple places.

**Comment by conniey** [comment]
@danroth27 The application that uses the library, I want to run on .NET Core. :) Thanks for the tip though.

**Comment by shaunluttin** [comment]
This issue also occurs with `.csproj` files.
